src/app/groups/[groupId]/expenses/expense-list.tsx

```diff
@@ -16,6 +17,51 @@ type Props = {
   groupId: string
 }
 
+const EXPENSE_GROUPS = {
+  THIS_WEEK: 'This week',
+  EARLIER_THIS_MONTH: 'Earlier this month',
+  LAST_MONTH: 'Last month',
+  EARLIER_THIS_YEAR: 'Earlier this year',
+  LAST_YEAR: 'Last year',
+  OLDER: 'Older',
+}
+
+function getDifferenceInDays(earlierDate: Date, laterDate: Date) {
+  const differenceInTime = laterDate.getTime() - earlierDate.getTime()
+  return Math.round(differenceInTime / (1000 * 3600 * 24)) // convert milliseconds to days
+}
+
+function getExpenseGroup(date: Dayjs, today: Dayjs) {
```

Comment:
Originally I had this:
<img width="623" alt="Screen Shot 2024-01-17 at 10 10 42" src="https://github.com/scastiel/spliit2/assets/11523186/086ab9ec-2cf0-4cac-b031-b8c295bfcf04">

But then I realized that the headers were then misleading so I have to actually check if the expense date is in the same week, or month, or year, etc. Because of that, I had to utilize the `dayjs` package